cfclient/cfapps/cache.go

```diff
@@ -69,6 +68,22 @@ func (c *Cache) Start() {
 	}()
 }
 
+func (c *Cache) updateIntances() {
+	c.isUpdating = true
+	var wg sync.WaitGroup
+	GetInstance().app.Log.Debug("Updating status of applications")
+	now := time.Now()
+	c.sync.RLock()
+	for _, v := range c.Collection {
+		wg.Add(1)
+		go v.UpdateInstancesWG(&wg)
+	}
+	c.sync.RUnlock()
```

Comment:
Could be there a problem if an app gets updated while it's being accessed from somewhere else? Here we release the lock while the update is still happening.